### PR TITLE
Clean up the code that builds uses a bit in LSRA.

### PR DESCRIPTION
**Pull request by pgavlin**
These cleanups are enabled by the recent changes to SIMD8 handling and
the removal of `fixedCandidateMask`.

### CODE REVIEW

**Comment by pgavlin** [comment]
@CarolEidt @dotnet/jit-contrib PTAL.

This is mostly for cleanliness, though it does produce a very modest reduction in instructions retired (less than a tenth of a percent).

File: src/jit/lsra.cpp [hunk 1]
```diff
@@ -3968,30 +3968,17 @@ void LinearScan::buildRefPositionsForNode(GenTree*                  tree,
...
+            Interval* const i = locInfo.interval;
             if (useNode->gtLsraInfo.isTgtPref)
             {
                 prefSrcInterval = i;
             }
 
-            const bool regOptionalAtUse = useNode->IsRegOptional();
-            const bool delayRegFree     = (hasDelayFreeSrc && useNode->gtLsraInfo.isDelayFree);
-
-            assert(isCandidateLocalRef(useNode) == i->isLocalVar);
```
Comment (CarolEidt) [inline src/jit/lsra.cpp]: Was it intentional that you removed this assert?

Comment (pgavlin) [inline src/jit/lsra.cpp]: It was: I didn't feel that it added much value as this sort of condition should be checked elsewhere. I can put it back if you like.

Comment (CarolEidt) [inline src/jit/lsra.cpp]: OK, I'm fine with removing it.

**Pull request merged**